src/cursor.js

```diff
@@ -289,6 +289,21 @@ export default class Cursor {
     return content.adoptElement(node, this.win.document)
   }
 
+  // Check whether the cursor's closest parent element node matches
+  // the given tagName (and optional className), and return that
+  // parent if it does or null if not.
+  getParentContainer (tagName, className) {
```

Comment:
This method only checks the direct parent. Usually in editable.js we collect all ancestors up to the host so you now inside which elements e.g. a cursor is currently in. Only checking the direct parent is not enough in some cases.

We already have `getTags()` and `getTagsByName()` which return all elements up to the host which affect a range.

`getTags()` and `getTagsByName()` are currently only available in the selection. But I think we could simply move them up to the cursor (The `Selection` class extends `Cursor`).

What do you think about this approach?

Reply:
@peyerluk, that would also work if it’s guaranteed that they both return an ordered list of matching ancestors. 

Does the performance difference between the two (fetching the closest matching ancestor vs. building a list of all matching ancestors) matter to you?

Reply:
The performance difference is negligible. 

The order will be reliable for a cursor. In case of the selection there are many cases of tags which surround the selection are contained within it or only start or end in the selection.

Reply:
Works for me then 👍🏼

Do you want to make the change, or would you like to close this PR in exchange for a new PR, or should I repurpose this PR?

Reply:
If you would like it sooner rather than later you can go ahead :) Otherwise I can do this the next time I'll do some coding in editable.js.

Reply:
@peyerluk, I hoisted 

https://github.com/livingdocsIO/editable.js/blob/59962fe995b2d7c3c0fdb4015ca9f70d0460dfca/src/selection.js#L244-L261

into [cursor.js](https://github.com/livingdocsIO/editable.js/blob/master/src/cursor.js) and that seems to have worked. With that change, I can now rewrite the new `getParentContainer()` function like so:
```javascript
  // Check whether the cursor's closest parent element node matches
  // the given tagName (and optional className), and return that
  // parent if it does or null if not.
  getParentContainer (tagName, className) {
    const tags = this.getTags()

    let node = tags.shift()
    if (node.nodeType !== Node.TEXT_NODE) return null

    if (tags.length === 0) return null

    node = tags.shift()
    if (node.nodeName !== tagName) return null
    if (className && !node.classList.contains(className)) return null

    return node
  }
```
Should probably be renamed to `getParentTag()` to fit in with other names in this class.

However, I have a feeling that you may not be too enthusiastic about this function because it may be too specific? Either way, happy to

 - either keep and expand this PR and this function with our discussed changes; or
 - close this PR and submit a new one that contains only hoisting the two functions.

Reply:
Addendum: when hoisting `getTags()` and `getTagsByName()` from `Selection` into `Cursor` I think I should change my other PR https://github.com/livingdocsIO/editable.js/pull/198/files accordingly and hoist `getTagNames()` into `Cursor` as well 🤔

Reply:
Addressed all of the above, still testing some more though.

Reply:
I like the hoisting of `getTags()` etc. to Cursor. But I think  the `getParentContainer()` method is too specific to your use case and should not be part of editable.js. This code can easily live in client apps once we expose `getTags()` on the `Cursor` class.

Reply:
Fairnuff. Want me to adjust this PR accordingly?